lib/babe/verify_integration_test.go

```diff
@@ -265,57 +294,74 @@ func TestVerificationManager_VerifyBlock_MultipleEpochs(t *testing.T) {
 	require.NoError(t, err)
 	slotState := state.NewSlotState(db)
-
 	verificationManager := NewVerificationManager(babeService.blockState, slotState, babeService.epochState)
 
-	bestBlockHash := babeService.blockState.BestBlockHash()
-	runtime, err := babeService.blockState.GetRuntime(bestBlockHash)
+	const epoch = uint64(0)
+	epochDescriptor, err := babeService.initiateEpoch(epoch)
 	require.NoError(t, err)
 
-	const futureEpoch = uint64(2)
-	err = babeService.epochState.(*state.EpochState).SetEpochDataRaw(futureEpoch, &types.EpochDataRaw{
-		Authorities: []types.AuthorityRaw{{
-			Key: [32]byte(keyring.Alice().(*sr25519.Keypair).Public().Encode()),
-		}},
-	})
-	require.NoError(t, err)
+	slot := Slot{
+		start:    getSlotStartTime(epochDescriptor.startSlot, babeService.constants.slotDuration),
+		duration: babeService.constants.slotDuration,
+		number:   epochDescriptor.startSlot,
+	}
 
-	futureEpochData, err := babeService.initiateEpoch(futureEpoch)
+	blockNumber01 := createTestBlockWithSlot(t, babeService, &genesisHeader, [][]byte{}, epochDescriptor, slot)
+	err = verificationManager.VerifyBlock(&blockNumber01.Header)
 	require.NoError(t, err)
 
-	futureEpochSlotNumber := int64(babeService.EpochLength()*futureEpoch+1) * 6
-	futureTimestamp := time.Unix(futureEpochSlotNumber, 0)
-
-	futureSlot := getSlot(t, runtime, futureTimestamp)
-	futureSlot.number = babeService.EpochLength()*futureEpoch + 1
-	block := createTestBlockWithSlot(t, babeService, &genesisHeader, [][]byte{}, futureEpoch, futureEpochData, futureSlot)
-
-	err = verificationManager.VerifyBlock(&block.Header)
+	err = babeService.blockState.(*state.BlockState).AddBlock(blockNumber01)
 	require.NoError(t, err)
 
-	epochData, err := babeService.initiateEpoch(0)
+	futureEpoch := uint64(1)
+	err = babeService.epochState.(*state.EpochState).StoreEpochDataRaw(futureEpoch, &types.EpochDataRaw{
+		Randomness: [32]byte{9},
+		Authorities: []types.AuthorityRaw{
+			{
+				Key:    [32]byte(keyring.Bob().(*sr25519.Keypair).Public().Encode()),
+				Weight: 1,
+			},
+			{
+				Key:    [32]byte(keyring.Alice().(*sr25519.Keypair).Public().Encode()),
+				Weight: 1,
+			},
+		},
+	})
 	require.NoError(t, err)
 
-	slot := getSlot(t, runtime, time.Now())
-	block = createTestBlockWithSlot(t, babeService, &genesisHeader, [][]byte{}, 0, epochData, slot)
+	futureEpochDescriptor, err := babeService.initiateEpoch(futureEpoch)
+	require.NoError(t, err)
 
-	err = verificationManager.VerifyBlock(&block.Header)
+	futureSlot := Slot{
+		start:    getSlotStartTime(futureEpochDescriptor.startSlot, babeService.constants.slotDuration),
+		duration: babeService.constants.slotDuration,
+		number:   futureEpochDescriptor.startSlot,
+	}
+	blockNumber02 := createTestBlockWithSlot(t, babeService,
+		&blockNumber01.Header, [][]byte{}, futureEpochDescriptor, futureSlot)
+	err = verificationManager.VerifyBlock(&blockNumber02.Header)
 	require.NoError(t, err)
+
+	// TODO: include test to verify skipped epoch
```

Comment:
is this still todo?

Reply:
it is already addressed here https://github.com/ChainSafe/gossamer/pull/3882